.claude/skills/membrane-config/scripts/validate_config.py

```
@@ -0,0 +1,145 @@
+            if parent == d:
+                break
+            d = parent
+    return None
+
+
+def download_schema():
+    os.makedirs(CACHE_DIR, exist_ok=True)
+    cached = os.path.join(CACHE_DIR, "v7.2.3.json")
+    if not os.path.isfile(cached):
+        print(f"Downloading schema from {PUBLISHED_SCHEMA_URL} ...", file=sys.stderr)
+        urllib.request.urlretrieve(PUBLISHED_SCHEMA_URL, cached)
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)